lib/util/SortableSet.js

```diff
@@ -50,16 +75,14 @@ class SortableSet extends Set {
 		this._invalidateCache();
 	}
 
-	/**
-	 * @returns {void}
-	 */
 	sort() {
 		this.sortWith(this._sortFn);
 	}
 
+	// eslint-disable-next-line valid-jsdoc
 	/**
-	 * @param {Function} fn - function to calculate value
-	 * @returns {TODO} - returns result of fn(this), cached until set changes
+	 * @param {(instance: SortableSet<T>) => TODO} fn function to calculate value
+	 * @returns {TODO} returns result of fn(this), cached until set changes
 	 */
 	getFromCache(fn) {
```

Comment:
This can use `@template R` for the return type.

Reply:
I think `fn` should actually be `function(SortableSet<T>): T`, and the return type of getFromCache should be `T`. That's because the type of `data` is `T` because the type of _cache is `Map<Function, T>`.